app/controllers/admin/api/cms/templates_controller.rb

```diff
@@ -104,7 +102,12 @@ def show
   ##~ op.parameters.add :name => "handler", :paramType => "query", :description => "text will be processed by the handler before rendering", :required => false, :allowableValues => { :valueType => "LIST", :values => ["textile", "markdown"]  }
   def update
     if @template.respond_to?(:section)
-      @template.section ||= find_section
+      section = find_section
+      @template.section = section if section.present?
+    end
+    if @template.respond_to?(:layout)
```

Comment:
CodeClimate complains about all `respond_to?` statements.

Maybe indeed we can refactor it to either:
- delegate the params update to the corresponding subclasses
- update the params in the controller, but before filter out the ones that shouldn't (or couldn't) be updated in the current instance (due to its class)

Reply:
Done. I refactored to remove all business logic from the controller and move it to its own services. I might have over-refactored 🫠